lib/result/object/TingMarcResult.php

```diff
@@ -36,22 +36,21 @@ protected function process() {
     $object = $this->result->searchResponse->result->searchResult[0]->collection->object[0];
     $records = $object->collection->record;
 
-    // If we have multiple records we need to figure out which one to use, by
-    // looking at the primaryObjectIdentifier. Otherwise, datafield will be a
-    // single object and we just use that.
+    // If we have multiple records we need to figure out which one to use.
     if (is_array($records)) {
       $primary_id = explode(':', $object->primaryObjectIdentifier->{'$'})[1];
       foreach ($records as $key => $record) {
-        foreach ($record->datafield as $key => $datafield) {
-          if ($datafield->{'@tag'}->{'$'} == '001') {
-            foreach ($datafield->subfield as $key => $subfield) {
-              if ($subfield->{'@code'}->{'$'}== 'a' && $subfield->{'$'} == $primary_id) {
-                $data = $record->datafield;
-                break;
-              }
-            }
-          }
+        $type = $record->{'@type'}->{'$'};
+        // We want the full bibliographic post and not some incomplete main,
+        // section or volume post.
+        if (drupal_strtolower($type) === 'bibliographic') {
```

Comment:
Is there a reason to use a Drupal function here? The client should not be restricted to Drupal. I would think you would achieve the same result by calling `strtolower` as types containing UTF-8 characters will never be equal to `bibliographic` no matter how you treat them.

Reply:
Done 👍

Reply:
I also removed an unused left over variable from the old method of finding the correct marc record